src/wallet/wallet.cpp

```diff
@@ -1894,6 +1894,30 @@ TransactionError CWallet::FillPSBT(PartiallySignedTransaction& psbtx, bool& comp
         }
     }
 
+    // Figure out if any witness_utxos should be dropped
+    std::vector<unsigned int> to_drop;
+    for (unsigned int i = 0; i < psbtx.inputs.size(); ++i) {
+        const auto& input = psbtx.inputs.at(i);
+        int wit_ver;
+        std::vector<unsigned char> wit_prog;
+        if (input.witness_utxo.IsNull() || !input.witness_utxo.scriptPubKey.IsWitnessProgram(wit_ver, wit_prog)) {
+            // There's a non-segwit input or Segwit v0, so we cannot drop any witness_utxos
+            to_drop.clear();
+            break;
+        }
+        if (wit_ver == 0) {
+            // Segwit v0, so we cannot drop any witness_utxos
+            to_drop.clear();
+            break;
+        }
+        to_drop.push_back(i);
```

Comment:
833ecb8dec1a10e3bc1b8d5bcaf0b6839277f582: maybe only do this if there actually _is_ a `non_witness_utxo`? Otherwise you might as well use a `go_drop` bool and loop over all inputs after this loop. Then you also wouldn't need to track the index.

Reply:
Done